File: ios/XmtpReactNative.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(XmtpReactNative, NSObject)

RCT_EXTERN_METHOD(address:(RCTPromiseResolveBlock *)resolve
                  reject:(RCTPromiseRejectBlock *)reject)

RCT_EXTERN_METHOD(configure:(NSString *)env
                  privateKey:(NSString *)privateKey
                  resolve:(RCTPromiseResolveBlock *)resolve
                  reject:(RCTPromiseRejectBlock *)reject)

RCT_EXTERN_METHOD(newConversation:(NSString *)peerAddress
                  conversationId:(NSString *)conversationId
                  metadata:(NSDictionary *)metadata
                  resolve:(RCTPromiseResolveBlock *)resolve
                  reject:(RCTPromiseRejectBlock *)reject)

RCT_EXTERN_METHOD(listConversations:(RCTPromiseResolveBlock *)resolve
                  reject:(RCTPromiseRejectBlock *)reject)

RCT_EXTERN_METHOD(listMessages:(NSString *)topic
                  resolve:(RCTPromiseResolveBlock *)resolve
                  reject:(RCTPromiseRejectBlock *)reject)

RCT_EXTERN_METHOD(sendMessage:(NSString *)topic
                  text:(NSString *)text
                  resolve:(RCTPromiseResolveBlock *)resolve
                  reject:(RCTPromiseRejectBlock *)reject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/XmtpReactNative.swift
import Foundation

// TODO: implement this using xmtp-ios

@objc(XmtpReactNative)
class XmtpReactNative: NSObject {

    @objc
    func configure(_ env: String,
                   privateKey: String,
                   resolve: @escaping RCTPromiseResolveBlock,
                   reject: @escaping RCTPromiseRejectBlock)
    {
        reject("unimplemented", "configure() is not implemented", nil)
    }

    @objc
    func newConversation(_ peerAddress: String,
                         conversationId: String,
                         metadata: NSDictionary,
                         resolve: @escaping RCTPromiseResolveBlock,
                         reject: @escaping RCTPromiseRejectBlock)
    {
        reject("unimplemented", "newConversation() is not implemented", nil)
    }

    @objc
    func listConversations(_ resolve: @escaping RCTPromiseResolveBlock,

                           reject: @escaping RCTPromiseRejectBlock)
    {
        reject("unimplemented", "listConversations() is not implemented", nil)
    }

    @objc
    func listMessages(_ topic: String,
                      resolve: @escaping RCTPromiseResolveBlock,
                      reject: @escaping RCTPromiseRejectBlock)
    {
        reject("unimplemented", "listMessages() is not implemented", nil)
    }

    @objc
    func sendMessage(_ topic: String, text: String,
                     resolve: @escaping RCTPromiseResolveBlock,
                     reject: @escaping RCTPromiseRejectBlock)
    {
        reject("unimplemented", "sendMessage() is not implemented", nil)
    }
}
